### PR TITLE
add retries for API calls

### PR DESCRIPTION
**Pull request by ddellacosta**
seems like we are killing the ETP voter guide API at the beginning when there are big bursts--adding some retries to hopefully slow it down a bit and make sure we don't lose anyone.

### CODE REVIEW

File: src/election_notification_works/handlers/voter_guide.clj [hunk 1]
```diff
@@ -32,12 +32,24 @@
...
-                  (http/get voter-guide-api-url))]
-    (if (= 200 (:status resp))
-      (:body resp)
-      (log/error "Got a non-200 response: " (pr-str resp)))))
+  ([zip]
+   (call-external-voter-guide-api zip 10))
+  ([zip tries]
+   (try-try-again
+    {:decay :exponential :tries tries}
+    (fn [zip']
```
Comment (danielglauser) [inline src/election_notification_works/handlers/voter_guide.clj]: I don't think you need `zip'` here, `zip` will be in scope within the closure.

Comment (danielglauser) [inline src/election_notification_works/handlers/voter_guide.clj]: Not a big deal though, what you have here should work.

File: src/election_notification_works/handlers/voter_guide.clj [hunk 2]
```diff
@@ -32,12 +32,24 @@
...
+                      (http/get voter-guide-api-url))]
+        (case (:status resp)
+          200
+          (:body resp)
+
+          500
+          (do (log/error "Got a 500 response, retrying: " (pr-str resp))
+              (throw (Exception. "Retry voter guide API call")))
+
+          (log/error "Got a non-200 or -500 response: " (pr-str resp)))))
```
Comment (danielglauser) [inline src/election_notification_works/handlers/voter_guide.clj]: There's no need to log here, you can use an [error hook](https://github.com/joegallo/robert-bruce/blob/master/test/robert/test/bruce.clj#L239).

Comment (danielglauser) [inline src/election_notification_works/handlers/voter_guide.clj]: https://github.com/joegallo/robert-bruce/blob/master/test/robert/test/bruce.clj#L239

Comment (danielglauser) [inline src/election_notification_works/handlers/voter_guide.clj]: No need, my bad. LGTM

Comment (ddellacosta) [inline src/election_notification_works/handlers/voter_guide.clj]: Per discussion in Slack think this may not apply here.

**Pull request merged**